src/hypothesis_testings.rs

```diff
@@ -22,13 +22,13 @@ impl MannWhitneyU {
     where
         X: Iterator<Item = T>,
         Y: Iterator<Item = T>,
-        T: Ord,
+        T: PartialOrd,
     {
         let mut vs = xs
             .map(|x| (x, Group::X))
             .chain(ys.map(|y| (y, Group::Y)))
             .collect::<Vec<_>>();
-        vs.sort();
+        vs.sort_by(|a, b| a.partial_cmp(&b).unwrap_or(Ordering::Equal));
```

Comment:
I think that, in general, it's not good to handle values that cannot be compared to each other as `Ordering::Equal` (for example, please see below code.
```rust
let mut xs = vec![0.0, 4.0, 2.0, std::f64::NAN, 1.0, std::f64::NAN, 3.0];
xs.sort_by(|a, b| a.partial_cmp(&b).unwrap_or(Ordering::Equal));
println!("{:?}", xs);
// => [0.0, 2.0, 4.0, NaN, 1.0, NaN, 3.0]  (this isn't an expected result I think)
```

Instead, you can use the `ordered-float` trait to make floating values comparable as follows.
```rust
use ordered_float::OrderedFloat;

let mut xs = vec![0.0, 4.0, 2.0, std::f64::NAN, 1.0, std::f64::NAN, 3.0];
xs.sort_by_key(|a| OrderedFloat(*a));
println!("{:?}", xs);
// => [0.0, 1.0, 2.0, 3.0, 4.0, NaN, NaN]
```